### ISSUE TITLE
[ENHANCEMENT REQUEST] :  Heart Disease Prediction [END 2 END] - Machine Learning

### ISSUE THREAD

**Issue by kinza7124** [UTSAVS26/PyVerse#1555]
### Title

**Add SMOTE for Class Balancing to Improve Model Accuracy**

### Enhancement Aim

The aim of this enhancement is to address class imbalance in the dataset by applying SMOTE (Synthetic Minority Over-sampling Technique) during the preprocessing stage. Class imbalance can negatively impact the model's ability to correctly predict the minority class, often resulting in poor recall, precision, and F1-score for that class.

By introducing SMOTE:

- We synthetically generate new samples for the minority class.

- The dataset becomes more balanced, leading to improved generalization of the model.

- The model can make more accurate and fair predictions for both classes.

This change is a small yet meaningful step toward boosting model robustness and fairness.

### Changes

- Import SMOTE from imblearn.over_sampling in the preprocessing section.

- Apply SMOTE to the feature and target data before train-test split:

- X_balanced, y_balanced = smote.fit_resample(X, y)

- Update class distribution visualization to reflect the new balanced data using seaborn countplot.

- Modify train-test split to use X_balanced and y_balanced instead of original X and y.

- Retrain the model (e.g., RandomForestClassifier) using the balanced dataset.

- Add classification report to show improved metrics after applying SMOTE:

- Use classification_report(y_test, y_pred) from sklearn.metrics.

### Screenshots 📷

**Class Distribution Before SMOTE:** 

<img width="786" height="687" alt="Image" src="https://github.com/user-attachments/assets/b6e2aa69-71d1-4323-8d96-c0e6f63c5ebd" />

**Applying SMOTE:**

<img width="640" height="102" alt="Image" src="https://github.com/user-attachments/assets/f4d1d5c2-2f38-48a6-8109-49e7b06e07e6" />

**Class Distribution After Applying SMOTE:** 

<img width="797" height="703" alt="Image" src="https://github.com/user-attachments/assets/2e9e0860-6dc2-4046-baa7-562c32c0bba8" />

**Previously this accuracy was 83.82%: 
Now its improved to 86.06%**

<img width="1250" height="183" alt="Image" src="https://github.com/user-attachments/assets/9c14f64d-a75f-442a-9e45-9214a207a4af" />

### Guidelines

- [x] I have read the guidelines
- [ ] I have the link to my latest merged PR

### Full Name

Kinza Afzal

### Participant Role

GSSOC

**Comment by kinza7124** [comment]
Pls Assign the issue to me

**Comment by TejasriMasina** [comment]
Could you please assign me this? I would love to contribute.

**Comment by UTSAVS26** [comment]
Hi @TejasriMasina you can raise new issues as this issue has been raised by @kinza7124, so i am gonna assign it to @kinza7124
